### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by fisker**
- Move `toPath` and `FilterStream` to utils.js
- Remove `ObjectTransform` class, there was another class extends it, but not any more
- Use really private property

### CODE REVIEW

File: utils.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export const toPath = urlOrPath => {
+	if (!urlOrPath) {
+		return urlOrPath;
+	}
+
+	if (urlOrPath instanceof URL) {
+		urlOrPath = urlOrPath.href;
+	}
+
+	return urlOrPath.startsWith('file://') ? fileURLToPath(urlOrPath) : urlOrPath;
```
Comment (sindresorhus) [inline utils.js:13]: I just realized, we shouldn't do `urlOrPath.startsWith('file://')`. Only the instanceof check. We should only accept `URL` or string path.

File: package.json [hunk 2]
```diff
@@ -23,8 +23,7 @@
 		"index.js",
 		"index.d.ts",
 		"gitignore.js",
-		"stream-utils.js",
-		"to-path.js"
+		"utils.js"
```
Comment (sindresorhus) [inline package.json:26]:
```suggestion
		"utilities.js"
```

File: utilities.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import {fileURLToPath} from 'node:url';
+import {Transform} from 'node:stream';
+
+export const toPath = urlOrPath => {
+	if (!urlOrPath) {
+		return urlOrPath;
+	}
+
+	return urlOrPath instanceof URL ? fileURLToPath(urlOrPath) : urlOrPath;
+};
```
Comment (sindresorhus) [inline utilities.js:10]:
```suggestion
export const toPath = urlOrPath => urlOrPath instanceof URL ? fileURLToPath(urlOrPath) : urlOrPath;
```

File: utilities.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+		super({objectMode: true});
+		this.#filter = filter;
+	}
+
+	_transform(data, encoding, callback) {
+		if (this.#filter(data)) {
+			this.push(data);
+		}
+
+		callback();
```
Comment (sindresorhus) [inline utilities.js:25]:
```suggestion
		callback(undefined, this.#filter(data) && data);
```

**Comment by fisker** [comment]
👍

**Comment by fisker** [comment]
https://github.com/nodejs/node/blob/37f1dd9ccdee8c54e1a3465f73e727eb87555fe5/lib/internal/streams/transform.js#L191

**Comment by fisker** [comment]
`this.#filter(data) && data` = `false`

**Comment by fisker** [comment]
Just saw there is a `transform` option.

**Pull request merged**